Framework/PythonInterface/mantid/api/src/Exports/IFunction.cpp

```diff
@@ -35,6 +36,32 @@ PyObject *getCategories(IFunction &self) {
   return registered;
 }
 
+/// An object for constructing a shared_ptr that won't ever delete its pointee
+class NoDeleting {
+public:
+  /// Does nothing
+  void operator()(void *) {}
+  /// Does nothing
+  void operator()(const void *) {}
+};
+
+//------------------------------------------------------------------------------------------------------
+/**
+* Something that makes IFunction return to python a composite function
+* for Product function, Convolution or
+* any similar superclass of composite function.
+* @param self :: Enables it to be called as a member function on the
+* IFunction class
+*/
+Mantid::API::CompositeFunction_sptr castToCompositeFunction(IFunction &self) {
```

Comment:
@KarlPalmen I think now that `castToCompositeFunction` is a dangerous method. Could you please have a think if it can be removed and instead `IFunction` would have 'composite' methods such as `__len__()`, `getFunction` which return 0 or throw exceptions if the function is not a `CompositeFunction`?